datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -94,8 +97,10 @@ struct ExternalSorter {
     expr: Arc<[PhysicalSortExpr]>,
     metrics: ExternalSorterMetrics,
     fetch: Option<usize>,
+    /// Reservation for in_mem_batches
     reservation: MemoryReservation,
-    partition_id: usize,
+    /// Reservation for in memory sorting of batches
```

Comment:
```suggestion
    /// Reservation for in memory sorting of batches, prior to spilling.
    /// Without this reservation, when the memory budget is exhausted
    /// it might not be possible to merge the in memory batches as part 
    /// of spilling.
```